### PR TITLE
Add annalyns-infiltration

### PR DESCRIPTION
**Pull request by BNAndras**
Towards #149.

The exemplar file will be shown to mentors during a mentoring discussion for this exercise as an example of a great solution at that point in the syllabus. Therefore, any and all nitpicks are appreciated keeping that in mind.

### CODE REVIEW

**Comment by BNAndras** [comment]
configlet's not going to be happy until we deal with the `functions` prerequisite. No concept exercises teach that at the moment so this will be an issue for the `cars-assemble` exercise as well. It might be easier to have some placeholder exercises for these prerequisites set up. Then you solicit PRs to fill those exercises with working implementations. Before the syllabus goes live, all those placeholders will need to be dealt with so students can unlock exercises as they go through the syllabus.

**Comment by 0xNeshi** [comment]
Will create stubs, great suggestion :heart:

File: exercises/concept/annalyns-infiltration/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+You have four tasks: to implement the logic for determining if the above actions are available based on the state of the three characters found in the forest and whether Annalyn's pet dog is present or not.
+
+## 1. Check if a fast attack can be made
+
+Define the `can_fast_attack()` function that takes a boolean value that indicates if the knight is awake.
+This function returns `true` if a fast attack can be made based on the state of the knight.
+Otherwise, returns `false`:
+
+```cairo
```
Comment (0xNeshi) [inline exercises/concept/annalyns-infiltration/.docs/instructions.md:31]:
``````suggestion
```rust
``````

There is no Cairo highlight atm

File: exercises/concept/annalyns-infiltration/.docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Introduction
+
+Booleans in Cairo are represented by the predeclared boolean type `bool`, which values can be either `true` or `false`.
+
+```cairo
+let speeding = true;   // boolean variable 'speeding' set to 'true'
+let has_error = false; // boolean variable 'hasError' set to 'false'
```
Comment (0xNeshi) [inline exercises/concept/annalyns-infiltration/.docs/introduction.md:7]:
```suggestion
let has_error = false; // boolean variable 'has_error' set to 'false' 
```

File: exercises/concept/annalyns-infiltration/.meta/exemplar.cairo [hunk 3]
```diff
@@ -0,0 +1,17 @@
+pub fn can_fast_attack(knight_awake: bool) -> bool {
+    return !knight_awake;
```
Comment (0xNeshi) [inline exercises/concept/annalyns-infiltration/.meta/exemplar.cairo:2]:
```suggestion
    !knight_awake
```
nit: returning a value _without_ using `return` is a more "Cairo" way, update the other returns as well

**Review by 0xNeshi — CHANGES_REQUESTED** [review]

**Comment by 0xNeshi** [comment]
This PR https://github.com/exercism/cairo/pull/156 should fix the configlet problem

**Comment by BNAndras** [comment]
So `configlet lint` apparently doesn't consider WIP exercises when it checks if a prerequisite is in the concepts array. If we set the status of `lucians-luscious-lasagna` to beta or active, that warning goes away. https://exercism.org/docs/building/tracks/config-json#h-metadata seems to suggest concept exercises won't show up until we set `concept_exercises` to true. However, that might be referring to the Learn tab that contains the syllabus. I've never worked on an Exercism syllabus.

**Comment by 0xNeshi** [comment]
> So configlet lint apparently doesn't consider WIP exercises when it checks if a prerequisite is in any exercise's concepts array.

That's what I figured as well.

> If we set the status of lucians-luscious-lasagna to beta or active, that warning goes away.

Maybe we should put all exercises to `wip` status for now, as there might be changes as we add them and I figure out how I'd like to organize the syllabus.

>  https://exercism.org/docs/building/tracks/config-json#h-metadata seems to suggest concept exercises won't show up until we set concept_exercises to true.

I think this is the case as well. We will enable the syllabus only once all the concept exercises are added.

**Review by 0xNeshi — APPROVED** [review]

**Pull request merged**